book/src/developers/designs/domain_join.md

```diff
@@ -0,0 +1,111 @@
+# Domain Join - Machine Accounts
+
+There are a number of features we have been considering that will required us to finally give
+in and support machine accounts also know as domain joining.
+
+## Feature Requirements
+
+### Limiting Unix Password Auth
+
+Currently unix password authentication is targetted as the method for sudo. Initial access to the
+machine should come from ssh keys (and in future, ctap2).
+
+In order to maintain compatability with LDAP style authentication, we allow "anonymous hosts" to
+retrieve ssh public keys, and then perform sudo authentication.
+
+This has the obvious caveat that anyone can stand up a machine that trusts a Kanidm instance. This
+presents a double edged sword:
+
+* The owner of the machine by configuring it to authenticate to Kanidm is stating it has full trust of the authentication decisions Kanidm makes.
+* Users of Kanidm may not realise that they are accessing a machine that is not potentially managed by their IT or other central authority.
+
+To prevent this, unix authentication should be configurable to prevent usage to anonymous
+machines. This will require the machine to present machine authentication credentials simultaneously
+with the user's credentials.
+
+### Requesting Cryptographic Credentials
+
+When a user logs in to a machine, it may be required that they can use that authentication to identify
+themself to other systems. When a user authenticates with credentials such as ssh-keys, it's not
+possible to use these to request other forwardable credentials - and ssh agent forwarding only allows
+forwarding of ssh credentials, not other types of credentials that may be needed.
+
+In this case when a user authenticates with ssh, since we are a trusted machine we should be able to
+request short-term and limited credentials on the users behalf.
+
+An example is that we could dynamicly request TLS certificates or kerberos credentials.
+
+Normally with ssh in this manner, everything has to use kerberos. This would force users to kinit on
+their machine to ssh and forward their credentials to the next machine. This causes friction since
+configuring kerberos on machines is an exercise in frustration, and with BYOD it gets even worse. In
+addition if a use ssh's with an ssh key, the only viable kinit mechanism is password or password + totp. Both
```

Comment:
You've got a typo here.

Reply:
Could you be slightly more specific than "a typo here" while highlighting an entire paragraph? 😄

Reply:
I spotted the typo.